### PR TITLE
OADP-4357: AWS plugin required for all S3 storage

### PR DESCRIPTION
**Pull request by anarnold97**
### Jira

* [OADP-4357](https://issues.redhat.com/browse/OADP-4357)

### Version(s):

* OCP 4.13 → branch/enterprise-4.13
*  OCP 4.14 → branch/enterprise-4.14
* OCP 4.15 → branch/enterprise-4.15
* OCP 4.16 → branch/enterprise-4.16
* OCP 4.17 → branch/enterprise-4.17

### Link to docs preview:

* MCG
    * [DPA 1.2 and earlier](https://79455--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.html#oadp-installing-dpa-1-2-and-earlier_installing-oadp-mcg)
    * [DPA 1.3](https://79455--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg.html#oadp-installing-dpa-1-3_installing-oadp-mcg)
* ODF
    * [DPA 1.2 and earlier](https://79455--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.html#oadp-installing-dpa-1-2-and-earlier_installing-oadp-ocs)
    * [DPA 1.3](https://79455--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.html#oadp-installing-dpa-1-3_installing-oadp-ocs)

### QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/oadp-installing-dpa-1-2-and-earlier.adoc [hunk 1]
```diff
@@ -281,37 +281,39 @@ spec:
   configuration:
     velero:
       defaultPlugins:
-        - kubevirt # <1>
-        - gcp # <2>
-        - csi # <3>
-        - openshift # <4>
-      resourceTimeout: 10m # <5>
+        - aws # <1>
```
Comment (PrasadJoshi12) [inline modules/oadp-installing-dpa-1-2-and-earlier.adoc:284]: Attach BSL spec has only one backupLocation which is GCP.  AWS plugin is not needed here

File: modules/oadp-installing-dpa-1-3.adoc [hunk 2]
```diff
@@ -298,41 +299,43 @@ spec:
...
       defaultPlugins:
-        - kubevirt # <2>
-        - gcp # <3>
-        - csi # <4>
-        - openshift # <5>
-      resourceTimeout: 10m # <6>
-    nodeAgent: # <7>
-      enable: true # <8>
-      uploaderType: kopia # <9>
+        - aws # <2>
```
Comment (PrasadJoshi12) [inline modules/oadp-installing-dpa-1-3.adoc:302]: Attach BSL spec has only one backupLocation. Only one plugin is required either AWS or GCP

File: modules/oadp-installing-dpa-1-2-and-earlier.adoc (hunk 1 above)
Comment (anarnold97) [reply modules/oadp-installing-dpa-1-2-and-earlier.adoc:284]: @PrasadJoshi12 - updated. Thanks

File: modules/oadp-installing-dpa-1-3.adoc (hunk 2 above)
Comment (anarnold97) [reply modules/oadp-installing-dpa-1-3.adoc:302]: @PrasadJoshi12 - updated. Thanks

**Review by shubham-pampattiwar — APPROVED** [review]

File: modules/oadp-installing-dpa-1-2-and-earlier.adoc [hunk 3]
```diff
@@ -234,39 +233,40 @@ spec:
...
-<3> Set this value to `false` if you want to disable the Restic installation. Restic deploys a daemon set, which means that Restic pods run on each working node. In OADP version 1.2 and later, you can configure Restic for backups by adding `spec.defaultVolumesToFsBackup: true` to the `Backup` CR. In OADP version 1.1, add `spec.defaultVolumesToRestic: true` to the `Backup` CR.
-<4> Specify on which nodes Restic is available. By default, Restic runs on all nodes.
-<5> Specify the region, following the naming convention of the documentation of your object storage server.
-<6> Specify the URL of the S3 endpoint.
-<7> If you do not specify this value, the default name, `{credentials}`, is used. If you specify a custom name, the custom name is used for the backup location.
-<8> Specify a bucket as the backup storage location. If the bucket is not a dedicated bucket for Velero backups, you must specify a prefix.
-<9> Specify a prefix for Velero backups, for example, `velero`, if the bucket is used for multiple purposes.
+<1> The `aws` plugin is mandatory for all S3 storage.
```
Comment (anarnold97) [inline modules/oadp-installing-dpa-1-2-and-earlier.adoc:260]:
```suggestion
<1> An object store plugin corresponding to your storage locations is required. For all S3 providers, the required plugin is `aws`. For {azure-short} and {gcp-short} object stores, the `azure` or `gcp` plugin is required.
```

File: modules/oadp-installing-dpa-1-3.adoc [hunk 4]
```diff
@@ -246,43 +246,44 @@ spec:
...
-<5> Set this value to `true` if you want to enable `nodeAgent` and perform File System Backup.
-<6> Enter `kopia` or `restic` as your uploader. You cannot change the selection after the installation. For the Built-in DataMover you must use Kopia. The `nodeAgent` deploys a daemon set, which means that the `nodeAgent` pods run on each working node. You can configure File System Backup by adding `spec.defaultVolumesToFsBackup: true` to the `Backup` CR.
-<7> Specify the nodes on which Kopia or Restic are available. By default, Kopia or Restic run on all nodes.
-<8> Specify the region, following the naming convention of the documentation of your object storage server.
-<9> Specify the URL of the S3 endpoint.
-<10> If you do not specify this value, the default name, `{credentials}`, is used. If you specify a custom name, the custom name is used for the backup location.
-<11> Specify a bucket as the backup storage location. If the bucket is not a dedicated bucket for Velero backups, you must specify a prefix.
-<12> Specify a prefix for Velero backups, for example, `velero`, if the bucket is used for multiple purposes.
+<2> The `aws` plugin is mandatory for all S3 storage.
```
Comment (anarnold97) [inline modules/oadp-installing-dpa-1-3.adoc:275]:
```suggestion
<2> An object store plugin corresponding to your storage locations is required. For all S3 providers, the required plugin is `aws`. For {azure-short} and {gcp-short} object stores, the `azure` or `gcp` plugin is required.
```

File: modules/oadp-installing-dpa-1-2-and-earlier.adoc [hunk 5]
```diff
@@ -301,8 +301,8 @@ spec:
           bucket: <bucket_name> # <10>
           prefix: <prefix> # <11>
 ----
-<1> Optional: The `kubevirt` plugin is used with {VirtProductName}.
-<2> Specify the default plugin for the backup provider, for example, `gcp`, if appropriate.
+<1> Specify the default plugin for the backup provider, for example, `gcp`, if appropriate. The `aws` plugin is mandatory for all S3 storage.
```
Comment (anarnold97) [inline modules/oadp-installing-dpa-1-2-and-earlier.adoc:304]:
```suggestion
<1> An object store plugin corresponding to your storage locations is required. For all S3 providers, the required plugin is `aws`. For {azure-short} and {gcp-short} object stores, the `azure` or `gcp` plugin is required.
```

**Comment by anarnold97** [comment]
@PrasadJoshi12 - Please can you review again, are you ok with these changes?
Thanks

**Review by sseago — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by PrasadJoshi12 — APPROVED** [review]

**Comment by anarnold97** [comment]
squashed ready for merge review

**Comment by skrthomas** [comment]
merge review lgtm

**Pull request merged**